src/NUnitFramework/framework/Internal/ThreadUtility.cs

```diff
@@ -230,5 +230,35 @@ private enum WM : uint
             CLOSE = 0x0010
         }
 #endif
+
+        /// <summary>Gets current principal of a thread</summary>
+        /// <returns><see cref="Thread.CurrentPrincipal"/>; null if <see cref="Thread.CurrentPrincipal"/> getter throws <see cref="PlatformNotSupportedException"/>.</returns>
```

Comment:
Same here. Please use `<summary>` since `<returns>` won't show up in intellisense, and please use a full sentence including an end mark. An example of a full sentence would be: `Gets <see cref="Thread.CurrentPrincipal"/> or <see langword="null" /> if the current platform does not support it.`